### DISCUSSION TITLE
limit clause and offset clause Type is missing

### DISCUSSION THREAD

**Discussion by liaohongxing** [diesel-rs/diesel#3316] [Q&A]
I am using VScode + Rust-Analyzer to learn Diesel

When I set up https://github.com/diesel-rs/diesel/tree/master/examples/mysql/getting_started_step_1, If the offset or limit clause and filter are used together, the type is lost and no automatic prompt is displayed.

How can I appear auto prompt

The code works correctly

![截图_20220911142846](https://user-images.githubusercontent.com/35255844/189515305-31103c70-89d5-4f19-b2c4-d7532732972b.png)

![截图_20220911143145](https://user-images.githubusercontent.com/35255844/189515385-1dfd7210-380d-461d-a356-3b2321ac4124.png)

**Comment by weiznich** [comment]
Please report such issus to the rust-analyzer team. Their implementation of rusts type system is incomplete and as far as I know they are interested in knowing about such cases.
From diesels side I cannot see what we could do to fix such behavior, especially in cases like `limit` that do not even involve complex trait bounds.
